dev-client/src/persistence/kvStorage.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {MMKVLoader} from 'react-native-mmkv-storage';
+
+// helpful things to know about MMKV:
+//
+// there are two layers of library involved with MMKV.
+//
+// there's the underlying cross-platform native library:
+//   https://github.com/Tencent/MMKV
+// and then there's the react native wrapper library:
+//   https://www.npmjs.com/package/react-native-mmkv-storage
+// there is also a competing react native wrapper library:
+//   https://www.npmjs.com/package/react-native-mmkv
+//
+// the library we use simplifies encryption by automatically
+// generating an encryption key and storing it on the keychain,
+// which we would have to do manually if we were using the other
+// library. at time of writing there is not a well-documented
+// migration story from one to the other.
+//
+// MMKV does not necessarily write to disk after every operation,
+// it will keep data in memory and only flush it when necessary
+// for performance reasons. I'm not sure if the RN API exposes a
+// way to manually flush to disk, there's a `clearMemoryCache`
+// method which presumably does this but haven't confirmed.
```

Comment:
Question: I assume it will flush on app close -- is preventing data loss if the app crashes the main impetus for considering flushing manually? Are there other use cases you have in mind?

Reply:
that's what i was thinking about!